### ISSUE TITLE
Implement fluent.bundle

### ISSUE THREAD

**Issue by Freebios** [projectfluent/fluent-kotlin#46]
Implement a port of the [fluent.bundle](https://github.com/projectfluent/fluent.js/tree/master/fluent-bundle) from [projectfluent/fluent.js](https://github.com/projectfluent/fluent.js/) to ease the usage of fluent-kotlin

**Comment by Freebios** [comment]
This would be our (my organization https://github.com/stelerio, or I) first Open Source contribution, but we are interested in contributing to the Fluent project starting with the fluent.bundle implementation if this is not already in progress !

**Comment by Pike** [comment]
I wonder if helping along @JohnnyJayJay on https://github.com/JohnnyJayJay/javidaloca/ would be better.

**Comment by JohnnyJayJay** [comment]
@Pike That would certainly be productive for Kotlin on the JVM and thanks for mentioning this here. But as I understand, this particular issue is about Kotlin running on JavaScript, which requires a separate implementation.

**Comment by Freebios** [comment]
@JohnnyJayJay No I was considering a port in Kotlin, no JS binding / code here.
Implementing it in your project could be a solution, but I don't have experience in rust at all and it looks like, from a quick sight in your project, that it's already developed, right?

**Comment by JohnnyJayJay** [comment]
@Freebios
> No I was considering a port in Kotlin, no JS binding / code here.

I see. So the question is whether you only care about the JVM as a target or want to cover all Kotlin target platforms. The latter would require an independent Kotlin implementation of the spec from the ground up. Otherwise, it may indeed make sense to join efforts with me here.

> I don't have experience in rust at all and it looks like, from a quick sight in your project, that it's already developed, right?

Frankly, I'm not very knowledgable in Rust either and that is why I'm currently looking for more help with this project. The basic features are implemented and work, what's missing is a proper connection between the two code bases that allows for cross-compilation and automated building/deployment. That is, you currently have to compile the Rust code and make the resulting shared libraries available at runtime manually.

Other than that, some of the more advanced features are still lacking a Java API, namely custom types and functions. But that is not the priority right now and I'll need some more Rust expertise for that too.

So, long story short: feel free to get in touch with me if your aim is to make a Kotlin/JVM fluent library. I can give you more details about the current state of javidaloca or even discuss other options such as making a real implementation.

**Comment by Freebios** [comment]
@JohnnyJayJay 
Our goal is to expose a more user friendly API for our dev in Java so wether the implementation is in Kotlin or it's a bind from Rust implementation doesn't really matter at the end.
We would be glad to give you a hand on your project :+1: 

@Pike 
If there is no implementation for Kotlin planned at the moment I guess you can close this issue then.

**Comment by Pike** [comment]
@Freebios sounds great, thanks.

The intent of fluent-kotlin was really to open up Fluent to the numerous tooling efforts in the l10n ecosystem  that are written in Java.
